packages/@aws-cdk/aws-stepfunctions/README.md

```diff
@@ -508,6 +508,119 @@ new stepfunctions.StateMachine(stack, 'MyStateMachine', {
+
+### Read Permissions
+
+Grant `read` access to a state machine by calling the `grantRead()` API.
+
+```ts
+const role = new iam.Role(stack, 'Role', {
+  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+});
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  definition,
+});
+
+//give role read access to state machine
+stateMachine.grantRead(role);
+```
+
+The following read permissions are provided to a service principal by the `grantRead()` API:
+
+- `states:ListExecutions` - to state machine
+- `states:ListStateMachines` - to state machine
+- `states:DescribeExecution` - to executions
+- `states:DescribeStateMachineForExecution` - to executions
+- `states:GetExecutionHistory` - to executions
+- `states:ListActivities` - to `*`
+- `states:DescribeStateMachine` - to `*`
+- `states:DescribeActivity` - to `*`
+
+### Start Execution Permission 
+
+Grant permission to start an execution of a state machine by calling the `grantStartExecution()` API.
+
+```ts
+const role = new iam.Role(stack, 'Role', {
+  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+});
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  definition,
+});
+
+//give role permission to start execution of state machine
+stateMachine.grantStartExecution(role);
+```
+
+The following permission is provided to a service principal by the `grantStartExecution()` API:
+
+- `states:StartExecution` - to state machine
+
+### Task Response Permissions
+
+Grant permission to allow task responses to a state machine by calling the `grantTaskResponse()` API. 
+
+Task responses can be scoped down either to the state machine that holds a task that will use the callback pattern,
+or an activity task with a separate `ARN`.
+
+Grant permission to the state machine: 
+
+```ts
+const role = new iam.Role(stack, 'Role', {
+  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+});
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  definition,
+});
+
+//default arn is the state machine arn
+stateMachine.grantTaskResponse(role);
+```
+
+Grant permission to an activity task: 
+
+```ts
+const role = new iam.Role(stack, 'Role', {
+  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+});
+
+const activityArn = 'arn:aws:states:*:*:activity:ActivityPrefix*';
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  definition,
+});
+
+//specify the activity-level ARN for the task response
+stateMachine.grantTaskResponse(role, activityArn);
+```
+
+### Custom Permissions
+
+You can add any set of permissions to a state machine by calling the `grant()` API.
+
+```ts
+const user = new iam.User(stack, 'MyUser');
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  definition,
+});
+
+//give user permission to send task success to the state machine
+stateMachine.grant(user, ['states:SendTaskSuccess'], stateMachine.stateMachineArn);
```

Comment:
this feels like it should just be the default and a user should not have to explicitly pass in a state machine arn at all

Reply:
done